src/models/unit_tests/source_code_unit_test.rs

```diff
@@ -699,3 +699,77 @@ fn test_satisfies_outermost_enclosing_node() {
 
   assert!(source_code_unit.is_satisfied(*node, &rule_positive, &HashMap::new(), &mut rule_store,));
 }
+
+#[test]
+fn test_removes_blank_lines_after_inline_cleanup() {
+  let inline_cleanup_rule = piranha_rule! {
+    name= "inline_cleanup_rule",
+    query= "
+    (
+      (if_modifier
+          body : ((_) @body)
+          [
+            condition: (false)
+            condition: (parenthesized_statements (false))
+          ]
+      )@if_modifier
+    )          
+    ",
+    replace_node = "if_modifier",
+    replace = ""
+  };
+
+  let inline_rule = InstantiatedRule::new(&inline_cleanup_rule, &HashMap::new());
+
+  let source_code = r#"
+      def method_name
+        do_something if false
+      end
+  "#
+  .trim();
+
+  let piranha_arguments = PiranhaArgumentsBuilder::default()
+    .paths_to_codebase(vec![UNUSED_CODE_PATH.to_string()])
+    .language(PiranhaLanguage::from(RUBY))
+    .build();
+
+  let mut rule_store = RuleStore::new(&piranha_arguments);
+  let mut parser = piranha_arguments.language().parser();
+
+  let source_code_unit = SourceCodeUnit::new(
+    &mut parser,
+    source_code.to_string(),
+    &HashMap::new(),
+    PathBuf::new().as_path(),
+    &piranha_arguments,
+  );
+  let matches = source_code_unit.get_matches(
+    &inline_rule,
+    &mut rule_store,
+    source_code_unit.root_node(),
+    true,
+  );
+  assert_eq!(matches.len(), 1);
+  assert_eq!(
+    matches
+      .first()
+      .unwrap()
+      .associated_leading_empty_lines
+      .len(),
+    1
+  );
+  assert_eq!(
+    *matches
+      .first()
+      .unwrap()
+      .associated_leading_empty_lines
+      .first()
+      .unwrap(),
+    Range {
+      start_byte: 15,
+      end_byte: 24,
+      start_point: Point { row: 0, column: 15 },
+      end_point: Point { row: 1, column: 8 }
+    }
+  );
+}
```

Comment:
Hi @Bennet-Sunder ,

Thank you for your contribution! I tested this internally and noticed that the change might break some rewrites. The issue is that in some grammars (particularly Swift), whitespace and newlines are part of the node. If you delete them, you might accidentally delete parts of a sibling node unrelated to the node you're deleting.

Please accept this test suggestion to see what I mean. The test should fail with your change (make sure to add Swift at the top). My proposed solution is to delete only whitespace or newlines until you find the end_byte of the previous sibling node. Let me know if you need help with this.

```suggestion
}

#[test]
fn test_switch_entry_blank_lines() {
  let inline_cleanup_rule = piranha_rule! {
    name= "inline_cleanup_rule",
    query= "
    (
        (switch_entry
            (switch_pattern) @p1
            (switch_pattern) @p2
            (switch_pattern) @p3
            (switch_pattern) @p4
            (switch_pattern) @p5
            (switch_pattern) @p6
        ) @custom_entry
        (#eq? @p1 \".case_zeta_first\")
    )
    ",
    replace_node = "custom_entry",
    replace = ""
  };

  let inline_rule = InstantiatedRule::new(&inline_cleanup_rule, &HashMap::new());

  let source_code = r#"
    public var namespace: ParameterNamespace {
        switch self {
        case .case_random_word_alpha,
             .case_random_word_beta,
             .case_random_word_gamma,
             .case_random_word_delta:
            return .namespace_group_alpha
        case .case_zeta_first,
             .case_zeta_second,
             .case_zeta_third,
             .case_zeta_fourth,
             .case_zeta_fifth,
             .case_zeta_sixth:
            return .namespace_group_beta
        case .case_random_word_omega:
            return .namespace_group_gamma
        }
    }
  "#
  .trim();

  let piranha_arguments = PiranhaArgumentsBuilder::default()
    .paths_to_codebase(vec![UNUSED_CODE_PATH.to_string()])
    .language(PiranhaLanguage::from(SWIFT))
    .build();

  let mut rule_store = RuleStore::new(&piranha_arguments);
  let mut parser = piranha_arguments.language().parser();

  let mut source_code_unit = SourceCodeUnit::new(
    &mut parser,
    source_code.to_string(),
    &HashMap::new(),
    PathBuf::new().as_path(),
    &piranha_arguments,
  );

  source_code_unit.apply_rule(inline_rule, &mut rule_store, &mut parser, &None);
  let transformed_code = source_code_unit.code();

  let expected_code = r#"
    public var namespace: ParameterNamespace {
        switch self {
        case .case_random_word_alpha,
             .case_random_word_beta,
             .case_random_word_gamma,
             .case_random_word_delta:
            return .namespace_group_alpha
        case .case_random_word_omega:
            return .namespace_group_gamma
        }
    }
  "#
  .trim();
  assert_eq!(transformed_code, expected_code);
}

```